test/basic.js

```diff
@@ -84,6 +85,42 @@ tape('verify', function (t) {
   })
 })
 
+tape('rootHashes', function (t) {
+  t.plan(9)
+
+  var feed = create()
+  var evilfeed = create(feed.key, {secretKey: feed.secretKey})
+
+  feed.append('test', function (err) {
+    t.error(err, 'no error')
+
+    evilfeed.append('t\0st', function (err) {
+      t.error(err, 'no error')
+
+      var result = new Assembly()
+      result.on('add', function (ev) {
+        if (result.length === 2) {
+          t.notEqual(result[0], result[1])
+          t.equal(result[0].length, result[1].length)
+          t.notEqual(Buffer.compare(result[0][0].hash, result[1][0].hash), 0)
+        }
+      })
```

Comment:
this is a bit confusing to read. why not just name the `function (ev)` here and call it after doing result.push below? also removes the need to an additional dep.

``` js
var results = []

feed.rootHashes(0, onroot)
evilfeed.rootHashes(0, onroot)

function onroots (err, roots) {
  t.error(err, 'no error')
  t.ok(roots instanceof Array)
  result.push(roots)
  if (result.length < 2) return
  t.notEqual(result[0], result[1])
  t.equal(result[0].length, result[1].length)
  t.notEqual(Buffer.compare(result[0][0].hash, result[1][0].hash), 0)
}
```